src/ParallelHashMap.h

```diff
@@ -566,23 +565,55 @@ void ParallelHashMap<K,V>::insert(K key, V value)
     return;
 
   /* get lock hash */
-  #ifdef OPENMP
+#ifdef OPENMP
   size_t lock_hash = (std::hash<K>()(key) & (_table->bucket_count() - 1))
     % _num_locks;
 
   /* acquire lock */
   omp_set_lock(&_locks[lock_hash]);
-  #endif
+#endif
 
   /* insert value */
   _table->insert(key, value);
 
   /* release lock */
-  #ifdef OPENMP
+#ifdef OPENMP
+  omp_unset_lock(&_locks[lock_hash]);
+#endif
+}
+
+
+/**
+ * @brief Updates the value associated with a key in the parallel hash map.
+ * @details The thread first acquires the lock for the bucket associated with
```

Comment:
Reads well save for small typo: "acquires the lock for the bucket associated with the key is acquired" -> "acquires the lock for associated bucket"